engine/runtime/src/main/java/org/enso/interpreter/runtime/data/atom/AtomConstructor.java

```diff
@@ -77,16 +78,36 @@ public AtomConstructor(String name, ModuleScope definitionScope, Type type) {
    * @param builtin if true, the constructor refers to a builtin type (annotated with @BuiltinType
    */
   public AtomConstructor(String name, ModuleScope definitionScope, Type type, boolean builtin) {
+    this(name, definitionScope, type, builtin, false);
+  }
+
+  /**
+   * Creates a new Atom constructor for a given name. The constructor is not valid until {@link
+   * AtomConstructor#initializeFields} is called.
+   *
+   * @param name the name of the Atom constructor
+   * @param definitionScope the scope in which this constructor was defined
+   * @param type associated type
+   * @param builtin if true, the constructor refers to a builtin type (annotated with @BuiltinType
+   * @param isProjectPrivate if true, this constructor is project-private.
+   */
+  public AtomConstructor(
+      String name,
+      ModuleScope definitionScope,
+      Type type,
+      boolean builtin,
+      boolean isProjectPrivate) {
```

Comment:
Could we have this `isProjectPrivate` flag only in `Type` and let `AtomConstructor.isProjectPrivate()` delegate to `Type.isProjectPrivate()`?

Reply:
The flag would need to be renamed though, to `hasPrivateConstructors`.

Just to make it clear that the type being private _itself_ is different from a public type with private constructor.

Reply:
Ah, that field is already there 😅 Great

Reply:
Yes, that reflects the semantics better. Done in 640665b360e8d91ab75e9375e1c3bb338f125fce.